src/env-inl.h

```diff
@@ -95,6 +95,24 @@ v8::Local<v8::Object> AsyncHooks::native_execution_async_resource(size_t i) {
   return PersistentToLocal::Strong(native_execution_async_resources_[i]);
 }
 
+inline void AsyncHooks::SetJSPromiseHooks(v8::Local<v8::Function> init,
+                                          v8::Local<v8::Function> before,
+                                          v8::Local<v8::Function> after,
+                                          v8::Local<v8::Function> resolve) {
+  js_promise_hooks_[0].Reset(env()->isolate(), init);
+  if (!init.IsEmpty()) js_promise_hooks_[0].SetWeak();
```

Comment:
Why are these weak? I assume we’d want these handles to remain alive as long the Environment is?

Reply:
When I saw the vm-memleak failures, I thought that maybe these strong refs might be causing the error, so I added the fixup commit making them weak.

Of course, that didn't help, but I also thought that the contexts would hold them strongly when they're set via `SetPromiseHooks`, so their lifetimes would already be ensured.

Reply:
Right … let’s remove these `SetWeak()` lines here then, otherwise the code looks like they might be GC’ed when they actually won’t :)